pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,204 @@
+
+            * By running the [List pipelines](/docs/apis/rest-api/pipelines#list-pipelines) REST API query to obtain this value from `slug` in the response from the specific pipeline. For example:
+
+                ```bash
+                curl - X GET "https://api.buildkite.com/v2/organizations/{org.slug}/pipelines" \
+                  -H "Authorization: Bearer $TOKEN"
+                ```
+        - `REF` is usually replaced with `refs/heads/main` to enforce the IAM role's access and use to only the `main` branch, `refs/tags/*` to ensure only tagged releases are able to be deployed, or a wildcard `*` if the IAM role can access and use all branches.
+        - `BUILD_COMMIT` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+        - `STEP_KEY` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+
+    **Note:** When formulating your _subject_ field's value, you can replace any of the constituent field values above with a wildcard `*` to not set limits on those constituent fields.
+
+    You can also allow this IAM role to be used with other pipelines, branches, commits and steps by specifying multiple comma-separated values for the `agent.buildkite.com:sub` _subject_ field.
+
+    For example, to allow the IAM role to be used for any pipeline in the Buildkite organization `example-org`, when building on any branch or tag, specify a single _subject_ field value of `organization:example-org:pipeline:*:ref:refs:*`.
+
+1. Modify the `Condition` section's `IpAddress` values (`AGENT_PUBLIC_IP_ONE` and `AGENT_PUBLIC_IP_TWO`) with a list of your agent's IP addresses.
+
+    Only OIDC token exchange requests (for IAM roles) from Buildkite Agents with these IP addresses will be permitted.
+
+1. Verify that your custom trust policy is complete. The following example trust policy (noting that `AWS_ACCOUNT_ID` has not been specified) will only allow the exchange of an agent's OIDC tokens with IAM roles when:
+    * the Buildkite organization is `example-org`
+    * building on both the `main` branch and tagged releases
+    * on Buildkite Agents whose IP addresses are either `192.0.2.0` or `198.51.100.0`
+
+    ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        "agent.buildkite.com:aud": "sts.amazonaws.com"
+                    },
+                    "StringLike": {
```

Comment:
Does this need to be `ForAnyValue:StringLike`? Or is that if the *target* contains multiple keys?

Reply:
I _think_ that's just when there's multiple keys but now I'm not sure...

Reply:
Let's wait for more feedback on this one once the published docs are in use. We can always amend this at any point in the future if we need to.